exercises/luhn/example.go

```diff
@@ -1,40 +1,41 @@
 package luhn
 
-const testVersion = 1
+import "strings"
 
-func Valid(n string) bool {
-	if len(n) == 1 {
-		return false
-	}
-	d := extract(n)
-	if len(d) == 0 {
+const testVersion = 2
+
+func Valid(id string) bool {
+
+	if len(strings.TrimSpace(id)) == 1 {
 		return false
 	}
-	last := len(d) - 1
-	return (check(d[:last])+d[last])%10 == 0
-}
 
-func extract(n string) []int {
-	d := make([]int, 0, len(n))
-	for _, r := range n {
+	d := make([]int, 0, len(id))
+
+	for _, r := range id {
+		if r == ' ' {
+			continue
+		}
 		if r >= '0' && r <= '9' {
 			d = append(d, int(r-'0'))
+		} else {
```

Comment:
this makes me realize there are some alternatives for how to structure this. They all have some advantage or disadvantage, so you can pick one you like after trying out:

1. Don't bother having `if r == ' ' { continue }` above. Instead just have `else if r != ' ' { return false }` here. Pro: Fewer clauses. Cons: May be a little harder to understand in what cases it does neither.

2. Invert the checks, say `if r < '0' || r > '9' { return false }` above. Pro: Now the `d = append` can be unindented. Con: Have to read through two conditions before getting to the actual loop body